src/libasr/pass/pass_list_concat.cpp

```diff
@@ -0,0 +1,222 @@
+                        right_list: list[i32]) -> list[i32]:
+                    result_list: list[i32]
+                    result_list = []
+                    i: i32
+                    for i in range(len(left_list)):
+                        result_list.append(left_list[i])
+                    for i in range(len(right_list)):
+                        result_list.append(right_list[i])
+                    return result_list
+            */
+            SymbolTable* list_concat_symtab = al.make_new<SymbolTable>(global_scope);
+            std::string fn_name = "_lcompilers_list_concat";
+
+            Vec<ASR::expr_t*> arg_exprs;
+            arg_exprs.reserve(al, 2);
+
+            Vec<ASR::stmt_t*> body;
+            body.reserve(al, 4);
+
+            // Declare `left_list`
+            ASR::symbol_t* arg = ASR::down_cast<ASR::symbol_t>(
+                ASR::make_Variable_t(al, loc, list_concat_symtab,
+                s2c(al, "left_list"), ASR::intentType::In, nullptr, nullptr,
+                ASR::storage_typeType::Default, list_type,
+                ASR::abiType::Source, ASR::accessType::Public,
+                ASR::presenceType::Required, false));
+            ASR::expr_t* arg_expr = ASRUtils::EXPR(
+                ASR::make_Var_t(al, loc, arg));
+            arg_exprs.push_back(al, arg_expr);
+            list_concat_symtab->add_symbol("left_list", arg);
+
+            // Declare `right_list`
+            arg = ASR::down_cast<ASR::symbol_t>(
+                ASR::make_Variable_t(al, loc, list_concat_symtab,
+                s2c(al, "right_list"), ASR::intentType::In, nullptr, nullptr,
+                ASR::storage_typeType::Default, list_type,
+                ASR::abiType::Source, ASR::accessType::Public,
+                ASR::presenceType::Required, false));
+            arg_expr = ASRUtils::EXPR(ASR::make_Var_t(al, loc, arg));
+            arg_exprs.push_back(al, arg_expr);
+            list_concat_symtab->add_symbol("right_list", arg);
+
+            // Declare `right_list`
+            arg = ASR::down_cast<ASR::symbol_t>(
+                ASR::make_Variable_t(al, loc, list_concat_symtab,
+                s2c(al, "result_list"), ASR::intentType::Local, nullptr, nullptr,
+                ASR::storage_typeType::Default, list_type,
+                ASR::abiType::Source, ASR::accessType::Public,
+                ASR::presenceType::Required, false));
+            arg_expr = ASRUtils::EXPR(ASR::make_Var_t(al, loc, arg));
+            list_concat_symtab->add_symbol("result_list", arg);
+
+            // Empty List
+            ASR::expr_t* value = ASRUtils::EXPR(ASR::make_ListConstant_t(
+                al, loc, nullptr, 0, list_type));
+            // Initialize `result_list` with `empty value`
+            ASR::stmt_t* list_concat_stmt = ASRUtils::STMT(ASR::make_Assignment_t(
+                al, loc, arg_expr, value, nullptr));
+            body.push_back(al, list_concat_stmt);
+
+            // Declare `i` for iterations
```

Comment:
Please use `void PassUtils::create_idx_vars(Vec<ASR::expr_t*>& idx_vars, int n_dims, const Location& loc,
                             Allocator& al, SymbolTable*& current_scope, std::string suffix="_k");` for creating auxiliary index variables.

Reply:
I got it! Done.